libs/services/src/SlippageService/SlippageServiceMain.ts

```diff
@@ -15,35 +23,124 @@ export class SlippageServiceMain implements SlippageService {
+    order,
+    chainId,
+    baseTokenAddress,
+    quoteTokenAddress,
+  }: GetSlippageBpsParams) {
+    // Get the 5min standard deviation for the quote token (~288 points, 5min apart)
+    const [volatilityQuote, volatilityBase] = await Promise.all([
+      this.getVolatilityOnSettlement(chainId, quoteTokenAddress, order),
+      this.getVolatilityOnSettlement(chainId, baseTokenAddress, order),
     ]);
 
-    return Math.max(slippageQuoteToken, slippageBaseToken);
+    if (volatilityQuote === null || volatilityBase === null) {
+      return null;
+    }
+
+    return Math.max(volatilityQuote, volatilityBase);
   }
 
-  private async getSlippageForToken(tokenAddress: string): Promise<Bps> {
+  /**
+   * Get the volatility of the asset in some time (enough for a solver to execute a solvable order)
+   *
+   * @param chainId
+   * @param tokenAddress
+   *
+   * @returns volatility in decimal format
+   */
+  private async getVolatilityOnSettlement(
+    chainId: SupportedChainId,
+    tokenAddress: string,
+    order?: OrderForSlippageCalculation
+  ): Promise<number | null> {
     const prices = await this.usdRepository.getUsdPrices(
-      SupportedChainId.MAINNET,
+      chainId,
       tokenAddress,
-      'daily'
+      '5m'
     );
 
     if (!prices) {
-      return DEFAULT_SLIPPAGE_BPS;
+      return null;
     }
 
-    const todayPrice = prices[prices.length - 1].price;
-    const yesterdayPrice = prices[prices.length - 2].price;
-    const bps = Math.abs(
-      ((todayPrice - yesterdayPrice) / yesterdayPrice) * 10000
+    // Get price of the token
+    const usdPrice = await this.usdRepository.getUsdPrice(
+      chainId,
+      tokenAddress
+    );
+
+    if (!usdPrice) {
+      return null;
+    }
+
+    // Calculate the average of the prices (in USD)
+    const averagePrice =
+      prices.reduce((acc, price) => acc + price.price, 0) / prices.length;
+
+    // Calculate price differences between each price point and the average
+    const priceDifferences = prices.map((price) => price.price - averagePrice);
+
+    // Square the differences
+    const squaredDifferences = priceDifferences.map(
+      (difference) => difference ** 2
     );
 
-    return Math.ceil(bps);
+    // Calculate the variance
+    const variance =
+      squaredDifferences.reduce((acc, difference) => acc + difference, 0) /
+      prices.length;
+
+    // Calculate the standard deviation
+    const standardDeviation = Math.sqrt(variance);
+
+    // Average time between each data point
+    const averageTimeBetweenDataPoints =
+      (prices[prices.length - 1].date.getTime() - prices[0].date.getTime()) /
+      (prices.length - 1);
+
+    // Points in Time for Settlement
+    const pointsForFairSettlement =
+      FAIR_TIME_TO_SETTLEMENT / averageTimeBetweenDataPoints;
+
+    // Predict variance between now and a fair settlement
+    const volatilityForFairSettlement =
+      standardDeviation * Math.sqrt(pointsForFairSettlement);
```

Comment:
I don't get this one.
Can you explain why multiply standard dev by sqrt or points per settlement?

Reply:
This is the trick on how to use the variance to predict the volatility at any point in time

Here I calculate how many point points are needed between now and a fair time for settlement, and then I need to multiply the variance by sqrt of the number of points

If I want to predict whats the volatility in 15min, I need to multiply by `sqrt(3)` cause each point is ~5min --> so multiply by `1.7320508076`